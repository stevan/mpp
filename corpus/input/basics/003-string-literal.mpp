# Double-quoted strings
"hello world";
"";
"Hello, World!";
"Line 1\nLine 2";
"Tab\there";

# Single-quoted strings
'hello world';
'';
'It\'s a string';
'No interpolation: $var';

# Strings with escape sequences
"She said \"Hello\"";
"Path: C:\\Windows\\System32";
"Unicode: \x{263A}";
"\t\r\n";

# Mixed quotes
"It's easy";
'He said "Hello"';

# Special characters
"!@#$%^&*()";
"[]{}<>";
"+-*/=";

# Multi-word strings
"The quick brown fox jumps over the lazy dog";
"Lorem ipsum dolor sit amet";

# Strings with numbers
"123";
"3.14159";
"Version 1.0.0";

# Empty and whitespace
"";
" ";
"   ";
"\n";

# Path-like strings
"/usr/local/bin";
"C:\\Program Files";
"../relative/path";

# URL-like strings
"https://example.com";
"http://localhost:8080/api";