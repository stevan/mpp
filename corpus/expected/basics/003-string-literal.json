[
  {
    "type": "String",
    "value": "\"hello world\""
  },
  {
    "type": "String",
    "value": "\"\""
  },
  {
    "type": "String",
    "value": "\"Hello, World!\""
  },
  {
    "type": "String",
    "value": "\"Line 1\\nLine 2\""
  },
  {
    "type": "String",
    "value": "\"Tab\\there\""
  },
  {
    "type": "String",
    "value": "'hello world'"
  },
  {
    "type": "String",
    "value": "''"
  },
  {
    "type": "String",
    "value": "'It\\'s a string'"
  },
  {
    "type": "String",
    "value": "'No interpolation: $var'"
  },
  {
    "type": "String",
    "value": "\"She said \\\"Hello\\\"\""
  },
  {
    "type": "String",
    "value": "\"Path: C:\\\\Windows\\\\System32\""
  },
  {
    "type": "String",
    "value": "\"Unicode: \\x{263A}\""
  },
  {
    "type": "String",
    "value": "\"\\t\\r\\n\""
  },
  {
    "type": "String",
    "value": "\"It's easy\""
  },
  {
    "type": "String",
    "value": "'He said \"Hello\"'"
  },
  {
    "type": "String",
    "value": "\"!@#$%^&*()\""
  },
  {
    "type": "String",
    "value": "\"[]{}<>\""
  },
  {
    "type": "String",
    "value": "\"+-*/=\""
  },
  {
    "type": "String",
    "value": "\"The quick brown fox jumps over the lazy dog\""
  },
  {
    "type": "String",
    "value": "\"Lorem ipsum dolor sit amet\""
  },
  {
    "type": "String",
    "value": "\"123\""
  },
  {
    "type": "String",
    "value": "\"3.14159\""
  },
  {
    "type": "String",
    "value": "\"Version 1.0.0\""
  },
  {
    "type": "String",
    "value": "\"\""
  },
  {
    "type": "String",
    "value": "\" \""
  },
  {
    "type": "String",
    "value": "\"   \""
  },
  {
    "type": "String",
    "value": "\"\\n\""
  },
  {
    "type": "String",
    "value": "\"/usr/local/bin\""
  },
  {
    "type": "String",
    "value": "\"C:\\\\Program Files\""
  },
  {
    "type": "String",
    "value": "\"../relative/path\""
  },
  {
    "type": "String",
    "value": "\"https://example.com\""
  },
  {
    "type": "String",
    "value": "\"http://localhost:8080/api\""
  }
]